File: core/src/main/java/org/ecos/reader/core/io/files/FileImpl.java
package org.ecos.reader.core.io.files;

public class FileImpl implements File {
    private Path mPath;

    private FileImpl(Path path) {
        mPath = path;
    }

    public static File from(Path path){
        return new FileImpl(path);
    }

    @Override
    public Path getPath() {
        return mPath;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        FileImpl file = (FileImpl) o;

        return mPath.equals(file.mPath);
    }

    @Override
    public int hashCode() {
        return mPath.hashCode();
    }
}

File: core/src/main/kotlin/org/ecos/reader/core/io/files/FileImpl.kt
package org.ecos.reader.core.io.files

class FileImpl private constructor(private val mPath: Path) : File {
    override val path: Path
        get() = mPath

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false

        val file = other as FileImpl?

        return mPath == file!!.mPath
    }

    override fun hashCode(): Int {
        return mPath.hashCode()
    }

    companion object {

        fun from(path: Path): File {
            return FileImpl(path)
        }
    }
}